release: 2.69 (#15270)
canonical/snapd: NEWS.md

# New in snapd 2.69
* FDE: re-factor listing of the disks based on run mode model and model to correctly resolve paths
* FDE: run snapd from snap-failure with the correct keyring mode
* Snap components: allow remodeling back to an old snap revision that includes components
* Snap components: fix remodel to a kernel snap that is already installed on the system, but not the current kernel due to a previous remodel.
* Snap components: fix for snapctl inputs that can crash snapd
* Snap components: LP: #2104933 workaround for classic 24.04/24.10 models that incorrectly specify core22 instead of core24
* Confdb (experimental): load ephemeral data when reading data via snapctl get
* Confdb (experimental): load ephermeral data when reading data via snap get
* Confdb (experimental): rename <plug>-view-changed hook to observe-view-<plug>
* Confdb (experimental): rename confdb assertion to confdb-schema
* Confdb (experimental): change operator grouping in confdb-control assertion
* Confdb (experimental): support pruning ephemeral data from snapd

> * AppArmor: fix encoding of time.Time by using omitzero instead of omitempty (on go 1.24+)

Reviewer: This isn't related to AppArmor either, just the way in which snapd generally encodes time.Time in json. So I think this should be moved down around like 27 and changed to something like this:

```suggestion
* Fix encoding of time.Time by using omitzero instead of omitempty (on go 1.24+)
```

Author: Fixed

# New in snapd 2.69
* FDE: re-factor listing of the disks based on run mode model and model to correctly resolve paths
* FDE: run snapd from snap-failure with the correct keyring mode
* Snap components: allow remodeling back to an old snap revision that includes components
* Snap components: fix remodel to a kernel snap that is already installed on the system, but not the current kernel due to a previous remodel.
* Snap components: fix for snapctl inputs that can crash snapd
* Confdb (experimental): load ephemeral data when reading data via snapctl get

> * Confdb (experimental): load ephermeral data when reading data via snap get

Reviewer 2: ```suggestion
* Confdb (experimental): load ephemeral data when reading data via snap get
```

# New in snapd 2.69
FDE: re-factor listing of the disks based on run mode model and model to correctly resolve paths

> FDE: run snapd with the correct keyring mode

Reviewer 3: I think this is specifially "from snap-failure", not in general,  double check with @valentindavid

Author: Thanks, fixed

Snap components: allow remodeling back to an old snap revision that includes components
Snap components: fix remodel to a kernel snap that is already installed on the system, but not the current kernel due to a previous remodel.
Snap components: fix for snapctl inputs that can crash snapd
Snap components: LP: #2104933 workaround for classic 24.04/24.10 models that incorrectly specify core22 instead of core24
Confdb (experimental): load ephemeral data when reading data via snapctl get
Confdb (experimental): load ephermeral data when reading data via snap get
Confdb (experimental): rename <plug>-view-changed hook to observe-view-<plug>
Confdb (experimental): rename confdb assertion to confdb-schema
Confdb (experimental): change operator grouping in confdb-control assertion
Confdb (experimental): support pruning ephemeral data from snapd

> AppArmor Prompting(experimental): simplify the listener

Reviewer: ```suggestion
AppArmor Prompting (experimental): simplify the listener
```

Author: Fixed

> AppArmor Prompting(experimental): use omitzero when encoding time.Time

Reviewer: This isn't actually specific to prompting, this change was made across the snapd codebase

Author: fixed as per below suggestion

> AppArmor Prompting(experimental): use omitzero when encoding time.Time

Reviewer: ```suggestion
Fix encoding of time.Time by using omitzero instead of omitempty
```

Author: Fixed

> AppArmor Prompting(experimental): extend the probed features to include the presence of files, as well as directories

Reviewer: This also isn't technically prompting-specific.

```suggestion
AppArmor: extend the probed features to include the presence of files, as well as directories
```

Author: Fixed

> AppArmor Prompting(experimental): make snap icons available over the snapd API without requiring network access at runtime

Reviewer: This also isn't prompting-specific (though it was motivated by prompting). I'd move it below and say something like:

```suggestion
Provide snap icon fallback for /v2/icons without requiring network access at runtime
```

Author: Fixed

AppArmor metadata tagging (disabled): probe parser support for tags
AppArmor metadata tagging (disabled): implement notification protocol v5
Confidential VMs: sysroot.mount is now dynamically created by snap-bootstrap instead of being a static file in the initramfs
Confidential VMs: Add new implementation of snap integrity API
Non-suid snap-confine: first phase to replace snap-confine suid with capabilities to achieve the required permissions

> Add eMMC gadget update support

Reviewer 3: do we have docs for this?

Author: I do not see anything, asking Philip...

Author: @pedronis, no not yet

> Support reexec when using /usr/libexec/snapd on the host (Fedora/Amazon Linux and Tumbleweed)

Reviewer 4: ```suggestion
Support reexec when using /usr/libexec/snapd on the host (Arch Linux, openSUSE)
```
Reexec on Fedora is not yet supported due to missing handling of a major LSM

Author: Fixed

Auto detect snap mount dir location on unknown distributions
Modify snap-confine AppArmor template to allow all glibc HWCAPS subdirectories to prevent launch errors 
Fix setting snapd permissions through permctl for openSUSE
Fix snap struct json tags typo
Fix snap pack configure hook permissions check incorrect file mode
Fix gadget snap reinstall to honor existing sizes of partitions
Fix to update command line when re-executing a snapd tool
Workaround for snapd-confine time_t size differences between architectures
Disallow pack and install of snapd, base and os with specific configure hooks
Drop udev build dependency that is no longer required and add missing systemd-dev dependency
Build snap-bootstrap with nomanagers tag to decrease size by 1MB

> Interfaces: gpio-chardev | add new interface

Reviewer 3: this is experimental and wip, either remark that or leave it out for now

Interfaces: polkit | support custom polkit rules
Interfaces: opengl | LP: #2088456 fix GLX on nvidia when xorg is confined by AppArmor
Interfaces: log-observe | add missing udev rule
Interfaces: hostname-control | fix call to hostnamectl in core24

> Interfaces: egl-driver-libs | support nvidia kernel components

Author: Removed

Interfaces: network-control | allow removing created network namespaces
Interfaces: scsi-generic | re-enable base declaration for scsi-generic plug

> Interfaces: cuda-driver-libs | support nvidia kernel components

Reviewer 3: same

> # New in snapd 2.69

Author: Build dependency fix was cherry-picked into 2.68.4 so is not repeated here.

* FDE: re-factor listing of the disks based on run mode model and model to correctly resolve paths
* FDE: run snapd from snap-failure with the correct keyring mode
* Snap components: allow remodeling back to an old snap revision that includes components
* Snap components: fix remodel to a kernel snap that is already installed on the system, but not the current kernel due to a previous remodel.
* Snap components: fix for snapctl inputs that can crash snapd
* Confdb (experimental): load ephemeral data when reading data via snapctl get
* Confdb (experimental): load ephemeral data when reading data via snap get
* Confdb (experimental): rename {plug}-view-changed hook to observe-view-{plug}
* Confdb (experimental): rename confdb assertion to confdb-schema
* Confdb (experimental): change operator grouping in confdb-control assertion

> * Confdb (experimental): add confdb-control API

Author: New,

Also removed previous prune entry because it was reverted.

* AppArmor: extend the probed features to include the presence of files, as well as directories
* AppArmor prompting (experimental): simplify the listener
* AppArmor metadata tagging (disabled): probe parser support for tags

> * AppArmor metadata tagging (disabled): implement notification protocol v5

Author: OliverC bugfix for caching is incorporated in this work, which is not yet release.

* Confidential VMs: sysroot.mount is now dynamically created by snap-bootstrap instead of being a static file in the initramfs
* Confidential VMs: Add new implementation of snap integrity API
* Non-suid snap-confine: first phase to replace snap-confine suid with capabilities to achieve the required permissions
* Initial changes for dynamic security profiles updates
* Provide snap icon fallback for /v2/icons without requiring network access at runtime
* Add eMMC gadget update support
* Support reexec when using /usr/libexec/snapd on the host (Arch Linux, openSUSE)
* Auto detect snap mount dir location on unknown distributions
* Modify snap-confine AppArmor template to allow all glibc HWCAPS subdirectories to prevent launch errors

> * LP: #2102456 update secboot to bf2f40ea35c4 and modify snap-bootstrap to remove usage of go templates to reduce size by 4MB

Author: New

> * Fix snap-bootstrap to mount kernel snap from /sysroot/writable/system-data

Author: New

> * LP: #2106121 fix snap-bootstrap busy loop

Author: New

Reviewer 3: we are still waiting to land the fix for this no?

https://github.com/canonical/snapd/pull/15287

just to clarify

Author: correct, above in description the cherry pick indicates "(to be added)"

* Fix encoding of time.Time by using omitzero instead of omitempty (on go 1.24+)
* Fix setting snapd permissions through permctl for openSUSE
* Fix snap struct json tags typo
* Fix snap pack configure hook permissions check incorrect file mode
* Fix gadget snap reinstall to honor existing sizes of partitions
* Fix to update command line when re-executing a snapd tool

> * Fix 'snap validate' of specific missing newline and add error on missed case of 'snap validate --refresh' without another action

Author: New
